src/persist/src/operators/stream.rs

```diff
@@ -41,46 +50,215 @@ where
 {
     fn persist(
         &self,
-        token: (StreamWriteHandle<K, V>, StreamReadHandle<K, V>),
+        name: &str,
+        write: StreamWriteHandle<K, V>,
+    ) -> (
+        Stream<G, ((K, V), u64, isize)>,
```

Comment:
frank has mentioned a couple times that it's also appropriate for us to emit a single stream of Result instead of an ok and err stream. fyi in case that ends up making anything easier for you